pkg/fleet/installer/packages/datadogagent/interface_windows.go

```diff
@@ -303,6 +303,18 @@ func installAgentPackage(env *env.Env, target string, args []string, logFileName
 	}
 	logFile := path.Join(tempDir, logFileName)
 
+	// Stop the Datadog Agent services before running MSI command
+	log.Infof("stopping the datadogagent service")
+	err = winutil.StopService("datadogagent")
```

Comment:
the service should be stopped before `removeAgentIfInstalled` is called. I think `installAgentPackage` is always called after that. It might even make sense to just add it into `removeAgentIfInstalled`, I can't think of a scenario where we wouldn't want to stop the services, the MSI is always going to try.